### ISSUE TITLE
manual satin command 

### ISSUE THREAD

**Issue by alpharesearch** [Embroidermodder/Embroidermodder#70]
I was reading about this manual satin command in some comments here from 2014, at the time it looked like it should have been pushed upstream soon. I can not find it or was it renamed? And if it exists how to use it? Thanks for your help.

**Comment by Metallicow** [comment]
It is a work-in-progress (WIPz) Thank you for interest.

**Comment by alpharesearch** [comment]
Thanks for your answer, and I hope its OK to ask a more general question about embroidery here. 

Is there any open source or free way to use a custom TTF and print one word and have it create a satin pattern? 

I just ordered a Brother embroidery machine for my wife and I was looking for software and found that free embroidery software is very spares. On Google yours is ranked number one and the only other one I found was Thread that can be compiled via winelib. But Thread has not TTF support.

I did see the Inkscape plugin but this looks like it can only save polylines. If there is nothing else my guess would be that using Inkscape with some custom plugin would be the fastest way to hack something together.

**Comment by Metallicow** [comment]
The core members may be able to help you a bit more with this as they do the main development and are more familiar with the vector and typography end of things. I mainly tinker with the python and porting/experimental ends. The core members may have some more updated working code floating around that may/maynot help you with your endeavor. Please get in touch with them(You can use github @ mentions here).

The fastest way to reach a goal is often a hack, but writing code with the endgoal in mind and submitting it(even if it is just a starting concept/prototype) is often better as understanding of the exact purposes/goals in mind is better and gets results quicker, as more minds tackle a particular problem openly(such is the idea of open source). Either that or try to contract some work done with the core or minor members or thru our website donations normally gets more attentions and manpower set on specific goals faster.

The Inkscape plugs are IIRC still WIPz also but has seen more attention in certain circles, so you are on a good start.
Good luck and Stitchify!

**Comment by redteam316** [comment]
@alpharesearch,
We have done some work on filling TTF fonts with Satin (a very complex topic if you expect quality output), but haven't pushed anything upstream yet and do not have an ETA on it right now.

**Comment by alpharesearch** [comment]
Yes regarding Embroidermodder I did give the TTF satine filling some thought and at this point my guess would be first to load the font and manually specify for each letter how to do the stitching. This should also allow to specify some stitching on the beginning and end plus set point where the character begins and ends. After this step I guess this info could be saved and shared as long as it is not containing the TTF. Now users could just use the font with the extra information and from this point it would be very trivial and the satin itself is just a zigzag stitch?

Is your Satin option you are working on fully automatic or is there some manual labor required?

Regarding Inkscape I found the path effect editor, there is a stitch option and I found if I replace some parts in the CSV file after the export it creates something that at least looks like a satin stitch. But it requires to have two parallel path and it is quite a lot of manual labor to create it.

**Comment by redteam316** [comment]
Your on the right track with your thinking but it's a bit more involved than that. The TTF satin feature would also cover stitch density/scaling and other info pertaining to curves (bezier/etc), underlayment and various other things. This information saved off to a file format would allow us to then automatically handle overlaying the proper stitching to the TTF and at that point, we could also apply various effects such as italic, bold, kerning, etc...

I would say the TTF feature is fully automatic but initially it will likely be semi-auto since creating the stitch overlay files would require it to be done manually once or generated by other methods.

As far as editing the Inkscape CSV afterword, that would be a quite laborous. I've seen some various methods that people have used Inkscape to create a psuedo-satin fill, although again, none of them are exactly correct and don't handle underlayment either. Stuff like that is what we plan to incorporate into libembroidery at some point, it just hasn't happened yet.
